pkg/sidecar/sidecar_test.go

```diff
@@ -21,9 +26,284 @@ import (
+		cmdArgs string
+		// Should stdio be redirected? nil means the test's stdio
+		// will be passed through.
+		stdin  *os.File
+		stdout *os.File
+		stderr *os.File
+		// How long to run 'cmd' before checking results. If a timeout
+		// is set and expectTerminated is false, the command must NOT have
+		// exited before the timeout, so ensure it runs for long enough.
+		timeout time.Duration
+		// Is this command expected to exit before the timeout (or at all?)
+		// If false, ensure that the command runtime is long enough for it
+		// to still be running while the test checks run.
+		expectTerminated bool
+		// exit code expected if 'cmd' is expected to terminate. This
+		// is the raw OS exit code. These tests don't currently define
+		// a way to test for signal exits.
+		expectExitStatus int
+		expectSignalExit syscall.Signal
+		// Is this command expected to create a file? If so, the path
+		// to check for the file's existence.
+		expectFileExists string
+	}{
+		{
+			// Check that the command runs and exits, and
+			// we can observe a side effect to prove it ran
+			cmd:              "touch",
+			cmdArgs:          touchTestFile,
+			expectTerminated: true,
+			expectExitStatus: 0,
+			expectFileExists: touchTestFile,
+		},
+		{
+			// check nonzero exit code handling
+			cmd:              "false",
+			expectTerminated: true,
+			expectExitStatus: 1,
+		},
+		{
+			// run a sleep and wait for it to finish, ensuring that
+			// we actually wait for the exit of a process that takes
+			// a moment to run.
+			cmd:              "sleep",
+			cmdArgs:          "0.1",
+			expectTerminated: true,
+		},
+		{
+			// run a sleep and inspect state before it finishes
+			cmd:              "sleep",
+			cmdArgs:          "1",
+			expectTerminated: false,
+			timeout:          400 * time.Millisecond,
+		},
+		{
+			// signal exit handling - if a process exits with a signal
+			// we handle it gracefully
+			cmd:     "sh",
+			cmdArgs: "-c \"kill -TERM $$\"",
+			// Suppress the "signal: hangup" from the shell
+			stdout:           nil,
+			stderr:           nil,
+			expectTerminated: true,
+			expectSignalExit: syscall.SIGTERM,
+		},
+		{
+			// show that argument splitting is confusing - it will not handle single-quoted
+			// arguments how you might expect. This will fail with an unterminated string
+			// error (exit code 2) because the argument gets split into the vector
+			// ["-c", "'kill", "-TERM", "$$'"]. This shows that we
+			// should really be using an argument vector to accept
+			// commands, not a string.
+			cmd:     "sh",
+			cmdArgs: "-c 'kill -TERM $$'",
+			// Suppress the "TERM: 1: Syntax error: Unterminated
+			// quoted string" error message from the shell
+			stdout:           nil,
+			stderr:           nil,
+			expectTerminated: true,
+			expectExitStatus: 2,
+		},
+	}
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.cmd, func(t *testing.T) {
+			if runtime.GOOS == "windows" && tc.expectSignalExit != 0 {
```

Comment:
Is this extra check needed if you are already skipping windows at the beginning of the test?

Reply:
Probably not, now it's all skipped

Reply:
Fixed